addons/base_import/models.py

```diff
@@ -261,6 +263,112 @@ def _read_csv(self, record, options):
+        try:
+            thousand_separator = decimal_separator = False
+            for val in preview_values:
+                if val == '':
+                    continue
+                negative = False
+                # Careful that some countries use () for negative so replace it by - sign
+                if val.startswith('(') and val.endswith(')'):
+                    val = val.replace('(','',1).replace(')','',1)
+                    negative = True
+                # We also might have the currency symbol left or right from the value
+                val = self._remove_currency_symbol(cr, uid, val)
+                if val:
+                    val = '-'+val if negative else val
+                    if options.get('float_thousand_separator', False) and options.get('float_decimal_separator', False):
+                        val = val.replace(options.get('float_thousand_separator'), '').replace(options.get('float_decimal_separator'), '.')
+                    # We are now sure that this is a float, but we still need to find the 
+                    # thousand and decimal separator
+                    else:
+                        if val.count('.') > 1:
+                            options['float_thousand_separator'] = '.'
+                            options['float_decimal_separator'] = ','
+                        elif val.count(',') > 1:
+                            options['float_thousand_separator'] = ','
+                            options['float_decimal_separator'] = '.'
+                        elif val.find('.') > val.find(','):
+                            thousand_separator = ','
+                            decimal_separator = '.'
+                        elif val.find(',') > val.find('.'):
+                            thousand_separator = '.'
+                            decimal_separator = ','
+                else:
+                    # This is not a float so exit this try
+                    float('a')
+            if thousand_separator and not options.get('float_decimal_separator', False):
+                options['float_thousand_separator'] = thousand_separator
+                options['float_decimal_separator'] = decimal_separator
+            return ['float', 'monetary']
+        except ValueError:
+            pass
+        # Try to see if all values are a date or datetime
+        dt = datetime.datetime
+        separator = [' ','/','-']
+        date_patterns = options.get('date_format', False) and [options.get('date_format')] or []
+        if not date_patterns:
+            for sep in separator:
+                date_patterns.extend(["%d"+sep+"%m"+sep+"%Y", "%m"+sep+"%d"+sep+"%Y", "%Y"+sep+"%m"+sep+"%d", "%Y"+sep+"%d"+sep+"%m"])
+                date_patterns.extend(["%d"+sep+"%m"+sep+"%y", "%m"+sep+"%d"+sep+"%y", "%y"+sep+"%m"+sep+"%d", "%y"+sep+"%d"+sep+"%m"])
+        current_date_pattern = False
+        for date_pattern in date_patterns:
+            ok = True
```

Comment:
Couldn't this flag be better named? Also why not just set the field type list directly instead of setting a flag which sets the format?